llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -801,6 +801,22 @@ static void computeKnownBitsFromCond(const Value *V, Value *Cond,
 
   if (auto *Cmp = dyn_cast<ICmpInst>(Cond))
     computeKnownBitsFromICmpCond(V, Cmp, Known, SQ, Invert);
+
+  if (match(Cond, m_Not(m_Value(Cond))))
+    Invert = !Invert;
```

Comment:
Not in a br cond will be canonicalized away by swapping successors. Shouldn't need this.

Reply:
notice at least some regressions in opt-benchmark when I removed it,
one example I think is this pattern: 
```
  %630 = trunc i8 %629 to i1
  %not. = xor i1 %630, true
  %.0215.shrunk. = or i1 %spec.select472, %not.
  br i1 %.0215.shrunk., label %632, label %.thread
```

so I shall maybe try to make a test that cover this case?

Reply:
added the @test_not_trunc_cond_and that do not pass without this.

Reply:
Can you please split this off into a followup change? I think it makes sense to handle it, but if we do so, we should do it generically, not for trunc conditions in particular.